src/Compilers/CSharp/Portable/Binder/LookupResult.cs

```diff
@@ -191,7 +191,13 @@ internal static SingleLookupResult NotTypeOrNamespace(Symbol symbol, DiagnosticI
         internal static SingleLookupResult NotTypeOrNamespace(Symbol unwrappedSymbol, Symbol symbol, bool diagnose)
         {
             // TODO: determine correct diagnosis 
-            var diagInfo = diagnose ? new CSDiagnosticInfo(ErrorCode.ERR_BadSKknown, unwrappedSymbol, unwrappedSymbol.GetKindText(), MessageID.IDS_SK_TYPE.Localize()) : null;
+            object sym = unwrappedSymbol;
+            // for CS0118 we dont want to display the type, but the name
+            if (unwrappedSymbol.Kind == SymbolKind.Parameter && diagnose)
+            {
+                sym = new FormattedSymbol(unwrappedSymbol, SymbolDisplayFormat.CSharpShortErrorMessageFormat.WithParameterOptions(SymbolDisplayParameterOptions.IncludeName));
+            }
+            var diagInfo = diagnose ? new CSDiagnosticInfo(ErrorCode.ERR_BadSKknown, sym, unwrappedSymbol.GetKindText(), MessageID.IDS_SK_TYPE.Localize()) : null;
```

Comment:
This approach is used elsewhere and appears to work here:

```cs
        internal static SingleLookupResult NotTypeOrNamespace(Symbol unwrappedSymbol, Symbol symbol, bool diagnose)
        {
            var diagInfo = diagnose ? new CSDiagnosticInfo(ErrorCode.ERR_BadSKknown, unwrappedSymbol.Name, unwrappedSymbol.GetKindText(), MessageID.IDS_SK_TYPE.Localize()) : null;
            return new SingleLookupResult(LookupResultKind.NotATypeOrNamespace, symbol, diagInfo);
        }
```

(I just added `.Name`)